CRM/Civicase/Hook/PostProcess/ActivityFormStatusWordReplacement.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+use CRM_Civicase_Hook_Helper_CaseTypeCategory as CaseCategoryWordReplacementHelper;
+use CRM_Core_Session as Session;
+use function ts as civicaseTs;
+
+/**
+ * Activity Form Status Word Replacement.
+ *
+ * Replaces the "saved activity type" status message with one appropriate for the
+ * case category the activity belongs to.
+ */
+class CRM_Civicase_Hook_PostProcess_ActivityFormStatusWordReplacement {
+
+  /**
+   * Runs the status message word replacement.
+   *
+   * It clears the original status message, adds the word replacements for the
+   * activity's case category, and adds the replaced title. We keep the original
+   * status message text.
+   *
+   * @param String $formName
+   *   The Form class name.
+   * @param Object $form
+   *   The Form instance.
+   */
+  public function run($formName, $form) {
+    $caseId = $form->getVar('_caseId');
+
+    if (!$this->shouldRun($form)) {
+      return;
+    }
+
+    $caseCategoryName = CaseCategoryHelper::getCategoryName($caseId);
+    CaseCategoryWordReplacementHelper::addWordReplacements($caseCategoryName);
+    $translatedActivityTypeName = civicaseTs($form->getVar('_activityTypeName'));
+
+    // Gets and resets the status message from the activity form:
+    $statusMessages = Session::singleton()->getStatus(true);
+
+    Session::setStatus(
+      $statusMessages[0]['text'],
+      ts('%1 Saved', [1 => $translatedActivityTypeName]),
+      'success'
+    );
```

Comment:
Can we also fetch the status from the `statusMessages` variable instead of hardcoding it as success?
Also can we compare the title is what we expect before we replace it. That way this will stop working if something changes in core regarding the title and we will notice and fix it.